ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouter.java

```diff
@@ -235,7 +237,9 @@ public Future<GetBlobResult> getBlob(String blobIdStr, GetBlobOptions options, f
     }
     currentOperationsCount.incrementAndGet();
     final FutureResult<GetBlobResult> futureResult = new FutureResult<>();
-    GetBlobOptionsInternal internalOptions = new GetBlobOptionsInternal(options, false, routerMetrics.ageAtGet);
+    GetBlobOptionsInternal internalOptions =
+        new GetBlobOptionsInternal(options, options.getOperationType() == GetBlobOptions.OperationType.BlobChunkIds,
```

Comment:
Make a new constructor like this.

`  
GetBlobOptionsInternal(GetBlobOptions getBlobOptions,  NonBlockingRouterMetrics.AgeAtAccessMetrics ageAtAccessTracker) {
    this.getBlobOptions = getBlobOptions;
    this.getChunkIdsOnly = getBlobOptions.getOperationType() == GetBlobOptions.OperationType.BlobChunkIds;
    this.ageAtAccessTracker = ageAtAccessTracker;
  }
`

Reply:
1. If we use GetBlobOptionsInternal, I think set the getChunkIdsOnly outside of constructor would be more readable. Otherwise you will see something like new GetBlobOptionsInternal(options, false, routerMetrics.ageAtGet) but inside the constructor it set to true at some case. It's kind of not clear to me. 
2. As we discussed, I will have a follow pr to replace the GetBlobOptionsInternal with GetBlobOptions.

Reply:
> Otherwise you will see something like new GetBlobOptionsInternal(options, false, routerMetrics.ageAtGet) but inside the constructor it set to true at some case

Why will this happen ? They are two distinct ctors